frontend/javascripts/test/libs/task_pool.spec.js

```diff
@@ -0,0 +1,98 @@
+// @noflow
+import processTaskWithPool from "libs/task_pool";
+import * as Utils from "libs/utils";
+import test from "ava";
+
+test("processTaskWithPool should run a simple task", async t => {
+  t.plan(1);
+
+  const protocol = [];
+
+  const tasks = [
+    async () => {
+      await Utils.sleep(1);
+      protocol.push(1);
+    },
+  ];
+
+  await processTaskWithPool(tasks, 1);
+
+  t.is(protocol.length, 1);
+});
+
+test("processTaskWithPool should run tasks sequentially", async t => {
+  t.plan(1);
+
+  const protocol = [];
+
+  const tasks = [
+    async () => {
+      await Utils.sleep(1);
```

Comment:
This makes it more explicit. Otherwise the asserted result would also likely be the result if the tasks were ran in parallel.

```suggestion
      await Utils.sleep(3);
```